// chromadb/dist/chunk-KNSX3QQN.mjs
import {
  __esm
} from "./chunk-DANUECVT.mjs";

// ../../node_modules/.pnpm/@smithy+types@3.1.0/node_modules/@smithy/types/dist-es/middleware.js
var SMITHY_CONTEXT_KEY;
var init_middleware = __esm({
  "../../node_modules/.pnpm/@smithy+types@3.1.0/node_modules/@smithy/types/dist-es/middleware.js"() {
    "use strict";
    SMITHY_CONTEXT_KEY = "__smithy_context";
  }
});

// ../../node_modules/.pnpm/@smithy+types@3.1.0/node_modules/@smithy/types/dist-es/abort.js
var init_abort = __esm({
  "../../node_modules/.pnpm/@smithy+types@3.1.0/node_modules/@smithy/types/dist-es/abort.js"() {
    "use strict";
  }
});

// ../../node_modules/.pnpm/@smithy+types@3.1.0/node_modules/@smithy/types/dist-es/auth/auth.js
var HttpAuthLocation;
var init_auth = __esm({
  "../../node_modules/.pnpm/@smithy+types@3.1.0/node_modules/@smithy/types/dist-es/auth/auth.js"() {
    "use strict";
    (function(HttpAuthLocation2) {
      HttpAuthLocation2["HEADER"] = "header";
      HttpAuthLocation2["QUERY"] = "query";
    })(HttpAuthLocation || (HttpAuthLocation = {}));
  }
});

// ../../node_modules/.pnpm/@smithy+types@3.1.0/node_modules/@smithy/types/dist-es/auth/HttpApiKeyAuth.js
var HttpApiKeyAuthLocation;
var init_HttpApiKeyAuth = __esm({
  "../../node_modules/.pnpm/@smithy+types@3.1.0/node_modules/@smithy/types/dist-es/auth/HttpApiKeyAuth.js"() {
    "use strict";
    (function(HttpApiKeyAuthLocation2) {
      HttpApiKeyAuthLocation2["HEADER"] = "header";
      HttpApiKeyAuthLocation2["QUERY"] = "query";
    })(HttpApiKeyAuthLocation || (HttpApiKeyAuthLocation = {}));
  }
});

// ../../node_modules/.pnpm/@smithy+types@3.1.0/node_modules/@smithy/types/dist-es/auth/HttpAuthScheme.js
var init_HttpAuthScheme = __esm({
  "../../node_modules/.pnpm/@smithy+types@3.1.0/node_modules/@smithy/types/dist-es/auth/HttpAuthScheme.js"() {
    "use strict";
  }
});

// ../../node_modules/.pnpm/@smithy+types@3.1.0/node_modules/@smithy/types/dist-es/auth/HttpAuthSchemeProvider.js
var init_HttpAuthSchemeProvider = __esm({
  "../../node_modules/.pnpm/@smithy+types@3.1.0/node_modules/@smithy/types/dist-es/auth/HttpAuthSchemeProvider.js"() {
    "use strict";
  }
});

// ../../node_modules/.pnpm/@smithy+types@3.1.0/node_modules/@smithy/types/dist-es/auth/HttpSigner.js
var init_HttpSigner = __esm({
  "../../node_modules/.pnpm/@smithy+types@3.1.0/node_modules/@smithy/types/dist-es/auth/HttpSigner.js"() {
    "use strict";
  }
});

// ../../node_modules/.pnpm/@smithy+types@3.1.0/node_modules/@smithy/types/dist-es/auth/IdentityProviderConfig.js
var init_IdentityProviderConfig = __esm({
  "../../node_modules/.pnpm/@smithy+types@3.1.0/node_modules/@smithy/types/dist-es/auth/IdentityProviderConfig.js"() {
    "use strict";
  }
});

// ../../node_modules/.pnpm/@smithy+types@3.1.0/node_modules/@smithy/types/dist-es/auth/index.js
var init_auth2 = __esm({
  "../../node_modules/.pnpm/@smithy+types@3.1.0/node_modules/@smithy/types/dist-es/auth/index.js"() {
    "use strict";
    init_auth();
    init_HttpApiKeyAuth();
    init_HttpAuthScheme();
    init_HttpAuthSchemeProvider();
    init_HttpSigner();
    init_IdentityProviderConfig();
  }
});

// ../../node_modules/.pnpm/@smithy+types@3.1.0/node_modules/@smithy/types/dist-es/blob/blob-payload-input-types.js
var init_blob_payload_input_types = __esm({
  "../../node_modules/.pnpm/@smithy+types@3.1.0/node_modules/@smithy/types/dist-es/blob/blob-payload-input-types.js"() {
    "use strict";
  }
});

// ../../node_modules/.pnpm/@smithy+types@3.1.0/node_modules/@smithy/types/dist-es/checksum.js
var init_checksum = __esm({
  "../../node_modules/.pnpm/@smithy+types@3.1.0/node_modules/@smithy/types/dist-es/checksum.js"() {
    "use strict";
  }
});

// ../../node_modules/.pnpm/@smithy+types@3.1.0/node_modules/@smithy/types/dist-es/client.js
var init_client = __esm({
  "../../node_modules/.pnpm/@smithy+types@3.1.0/node_modules/@smithy/types/dist-es/client.js"() {
    "use strict";
  }
});

// ../../node_modules/.pnpm/@smithy+types@3.1.0/node_modules/@smithy/types/dist-es/command.js
var init_command = __esm({
  "../../node_modules/.pnpm/@smithy+types@3.1.0/node_modules/@smithy/types/dist-es/command.js"() {
    "use strict";
  }
});

// ../../node_modules/.pnpm/@smithy+types@3.1.0/node_modules/@smithy/types/dist-es/connection/config.js
var init_config = __esm({
  "../../node_modules/.pnpm/@smithy+types@3.1.0/node_modules/@smithy/types/dist-es/connection/config.js"() {
    "use strict";
  }
});

// ../../node_modules/.pnpm/@smithy+types@3.1.0/node_modules/@smithy/types/dist-es/connection/manager.js
var init_manager = __esm({
  "../../node_modules/.pnpm/@smithy+types@3.1.0/node_modules/@smithy/types/dist-es/connection/manager.js"() {
    "use strict";
  }
});

// ../../node_modules/.pnpm/@smithy+types@3.1.0/node_modules/@smithy/types/dist-es/connection/pool.js
var init_pool = __esm({
  "../../node_modules/.pnpm/@smithy+types@3.1.0/node_modules/@smithy/types/dist-es/connection/pool.js"() {
    "use strict";
  }
});

// ../../node_modules/.pnpm/@smithy+types@3.1.0/node_modules/@smithy/types/dist-es/connection/index.js
var init_connection = __esm({
  "../../node_modules/.pnpm/@smithy+types@3.1.0/node_modules/@smithy/types/dist-es/connection/index.js"() {
    "use strict";
    init_config();
    init_manager();
    init_pool();
  }
});

// ../../node_modules/.pnpm/@smithy+types@3.1.0/node_modules/@smithy/types/dist-es/crypto.js
var init_crypto = __esm({
  "../../node_modules/.pnpm/@smithy+types@3.1.0/node_modules/@smithy/types/dist-es/crypto.js"() {
    "use strict";
  }
});

// ../../node_modules/.pnpm/@smithy+types@3.1.0/node_modules/@smithy/types/dist-es/encode.js
var init_encode = __esm({
  "../../node_modules/.pnpm/@smithy+types@3.1.0/node_modules/@smithy/types/dist-es/encode.js"() {
    "use strict";
  }
});

// ../../node_modules/.pnpm/@smithy+types@3.1.0/node_modules/@smithy/types/dist-es/endpoint.js
var EndpointURLScheme;
var init_endpoint = __esm({
  "../../node_modules/.pnpm/@smithy+types@3.1.0/node_modules/@smithy/types/dist-es/endpoint.js"() {
    "use strict";
    (function(EndpointURLScheme2) {
      EndpointURLScheme2["HTTP"] = "http";
      EndpointURLScheme2["HTTPS"] = "https";
    })(EndpointURLScheme || (EndpointURLScheme = {}));
  }
});

// ../../node_modules/.pnpm/@smithy+types@3.1.0/node_modules/@smithy/types/dist-es/endpoints/EndpointRuleObject.js
var init_EndpointRuleObject = __esm({
  "../../node_modules/.pnpm/@smithy+types@3.1.0/node_modules/@smithy/types/dist-es/endpoints/EndpointRuleObject.js"() {
    "use strict";
  }
});

// ../../node_modules/.pnpm/@smithy+types@3.1.0/node_modules/@smithy/types/dist-es/endpoints/ErrorRuleObject.js
var init_ErrorRuleObject = __esm({
  "../../node_modules/.pnpm/@smithy+types@3.1.0/node_modules/@smithy/types/dist-es/endpoints/ErrorRuleObject.js"() {
    "use strict";
  }
});

// ../../node_modules/.pnpm/@smithy+types@3.1.0/node_modules/@smithy/types/dist-es/endpoints/RuleSetObject.js
var init_RuleSetObject = __esm({
  "../../node_modules/.pnpm/@smithy+types@3.1.0/node_modules/@smithy/types/dist-es/endpoints/RuleSetObject.js"() {
    "use strict";
  }
});

// ../../node_modules/.pnpm/@smithy+types@3.1.0/node_modules/@smithy/types/dist-es/endpoints/shared.js
var init_shared = __esm({
  "../../node_modules/.pnpm/@smithy+types@3.1.0/node_modules/@smithy/types/dist-es/endpoints/shared.js"() {
    "use strict";
  }
});

// ../../node_modules/.pnpm/@smithy+types@3.1.0/node_modules/@smithy/types/dist-es/endpoints/TreeRuleObject.js
var init_TreeRuleObject = __esm({
  "../../node_modules/.pnpm/@smithy+types@3.1.0/node_modules/@smithy/types/dist-es/endpoints/TreeRuleObject.js"() {
    "use strict";
  }
});

// ../../node_modules/.pnpm/@smithy+types@3.1.0/node_modules/@smithy/types/dist-es/endpoints/index.js
var init_endpoints = __esm({
  "../../node_modules/.pnpm/@smithy+types@3.1.0/node_modules/@smithy/types/dist-es/endpoints/index.js"() {
    "use strict";
    init_EndpointRuleObject();
    init_ErrorRuleObject();
    init_RuleSetObject();
    init_shared();
    init_TreeRuleObject();
  }
});

// ../../node_modules/.pnpm/@smithy+types@3.1.0/node_modules/@smithy/types/dist-es/eventStream.js
var init_eventStream = __esm({
  "../../node_modules/.pnpm/@smithy+types@3.1.0/node_modules/@smithy/types/dist-es/eventStream.js"() {
    "use strict";
  }
});

// ../../node_modules/.pnpm/@smithy+types@3.1.0/node_modules/@smithy/types/dist-es/extensions/checksum.js
var AlgorithmId;
var init_checksum2 = __esm({
  "../../node_modules/.pnpm/@smithy+types@3.1.0/node_modules/@smithy/types/dist-es/extensions/checksum.js"() {
    "use strict";
    (function(AlgorithmId2) {
      AlgorithmId2["MD5"] = "md5";
      AlgorithmId2["CRC32"] = "crc32";
      AlgorithmId2["CRC32C"] = "crc32c";
      AlgorithmId2["SHA1"] = "sha1";
      AlgorithmId2["SHA256"] = "sha256";
    })(AlgorithmId || (AlgorithmId = {}));
  }
});

// ../../node_modules/.pnpm/@smithy+types@3.1.0/node_modules/@smithy/types/dist-es/extensions/defaultClientConfiguration.js
var init_defaultClientConfiguration = __esm({
  "../../node_modules/.pnpm/@smithy+types@3.1.0/node_modules/@smithy/types/dist-es/extensions/defaultClientConfiguration.js"() {
    "use strict";
    init_checksum2();
  }
});

// ../../node_modules/.pnpm/@smithy+types@3.1.0/node_modules/@smithy/types/dist-es/extensions/defaultExtensionConfiguration.js
var init_defaultExtensionConfiguration = __esm({
  "../../node_modules/.pnpm/@smithy+types@3.1.0/node_modules/@smithy/types/dist-es/extensions/defaultExtensionConfiguration.js"() {
    "use strict";
  }
});

// ../../node_modules/.pnpm/@smithy+types@3.1.0/node_modules/@smithy/types/dist-es/extensions/index.js
var init_extensions = __esm({
  "../../node_modules/.pnpm/@smithy+types@3.1.0/node_modules/@smithy/types/dist-es/extensions/index.js"() {
    "use strict";
    init_defaultClientConfiguration();
    init_defaultExtensionConfiguration();
    init_checksum2();
  }
});

// ../../node_modules/.pnpm/@smithy+types@3.1.0/node_modules/@smithy/types/dist-es/http.js
var FieldPosition;
var init_http = __esm({
  "../../node_modules/.pnpm/@smithy+types@3.1.0/node_modules/@smithy/types/dist-es/http.js"() {
    "use strict";
    (function(FieldPosition2) {
      FieldPosition2[FieldPosition2["HEADER"] = 0] = "HEADER";
      FieldPosition2[FieldPosition2["TRAILER"] = 1] = "TRAILER";
    })(FieldPosition || (FieldPosition = {}));
  }
});

// ../../node_modules/.pnpm/@smithy+types@3.1.0/node_modules/@smithy/types/dist-es/http/httpHandlerInitialization.js
var init_httpHandlerInitialization = __esm({
  "../../node_modules/.pnpm/@smithy+types@3.1.0/node_modules/@smithy/types/dist-es/http/httpHandlerInitialization.js"() {
    "use strict";
  }
});

// ../../node_modules/.pnpm/@smithy+types@3.1.0/node_modules/@smithy/types/dist-es/identity/apiKeyIdentity.js
var init_apiKeyIdentity = __esm({
  "../../node_modules/.pnpm/@smithy+types@3.1.0/node_modules/@smithy/types/dist-es/identity/apiKeyIdentity.js"() {
    "use strict";
  }
});

// ../../node_modules/.pnpm/@smithy+types@3.1.0/node_modules/@smithy/types/dist-es/identity/awsCredentialIdentity.js
var init_awsCredentialIdentity = __esm({
  "../../node_modules/.pnpm/@smithy+types@3.1.0/node_modules/@smithy/types/dist-es/identity/awsCredentialIdentity.js"() {
    "use strict";
  }
});

// ../../node_modules/.pnpm/@smithy+types@3.1.0/node_modules/@smithy/types/dist-es/identity/identity.js
var init_identity = __esm({
  "../../node_modules/.pnpm/@smithy+types@3.1.0/node_modules/@smithy/types/dist-es/identity/identity.js"() {
    "use strict";
  }
});

// ../../node_modules/.pnpm/@smithy+types@3.1.0/node_modules/@smithy/types/dist-es/identity/tokenIdentity.js
var init_tokenIdentity = __esm({
  "../../node_modules/.pnpm/@smithy+types@3.1.0/node_modules/@smithy/types/dist-es/identity/tokenIdentity.js"() {
    "use strict";
  }
});

// ../../node_modules/.pnpm/@smithy+types@3.1.0/node_modules/@smithy/types/dist-es/identity/index.js
var init_identity2 = __esm({
  "../../node_modules/.pnpm/@smithy+types@3.1.0/node_modules/@smithy/types/dist-es/identity/index.js"() {
    "use strict";
    init_apiKeyIdentity();
    init_awsCredentialIdentity();
    init_identity();
    init_tokenIdentity();
  }
});

// ../../node_modules/.pnpm/@smithy+types@3.1.0/node_modules/@smithy/types/dist-es/logger.js
var init_logger = __esm({
  "../../node_modules/.pnpm/@smithy+types@3.1.0/node_modules/@smithy/types/dist-es/logger.js"() {
    "use strict";
  }
});

// ../../node_modules/.pnpm/@smithy+types@3.1.0/node_modules/@smithy/types/dist-es/pagination.js
var init_pagination = __esm({
  "../../node_modules/.pnpm/@smithy+types@3.1.0/node_modules/@smithy/types/dist-es/pagination.js"() {
    "use strict";
  }
});

// ../../node_modules/.pnpm/@smithy+types@3.1.0/node_modules/@smithy/types/dist-es/profile.js
var IniSectionType;
var init_profile = __esm({
  "../../node_modules/.pnpm/@smithy+types@3.1.0/node_modules/@smithy/types/dist-es/profile.js"() {
    "use strict";
    (function(IniSectionType2) {
      IniSectionType2["PROFILE"] = "profile";
      IniSectionType2["SSO_SESSION"] = "sso-session";
      IniSectionType2["SERVICES"] = "services";
    })(IniSectionType || (IniSectionType = {}));
  }
});

// ../../node_modules/.pnpm/@smithy+types@3.1.0/node_modules/@smithy/types/dist-es/response.js
var init_response = __esm({
  "../../node_modules/.pnpm/@smithy+types@3.1.0/node_modules/@smithy/types/dist-es/response.js"() {
    "use strict";
  }
});

// ../../node_modules/.pnpm/@smithy+types@3.1.0/node_modules/@smithy/types/dist-es/retry.js
var init_retry = __esm({
  "../../node_modules/.pnpm/@smithy+types@3.1.0/node_modules/@smithy/types/dist-es/retry.js"() {
    "use strict";
  }
});

// ../../node_modules/.pnpm/@smithy+types@3.1.0/node_modules/@smithy/types/dist-es/serde.js
var init_serde = __esm({
  "../../node_modules/.pnpm/@smithy+types@3.1.0/node_modules/@smithy/types/dist-es/serde.js"() {
    "use strict";
  }
});

// ../../node_modules/.pnpm/@smithy+types@3.1.0/node_modules/@smithy/types/dist-es/shapes.js
var init_shapes = __esm({
  "../../node_modules/.pnpm/@smithy+types@3.1.0/node_modules/@smithy/types/dist-es/shapes.js"() {
    "use strict";
  }
});

// ../../node_modules/.pnpm/@smithy+types@3.1.0/node_modules/@smithy/types/dist-es/signature.js
var init_signature = __esm({
  "../../node_modules/.pnpm/@smithy+types@3.1.0/node_modules/@smithy/types/dist-es/signature.js"() {
    "use strict";
  }
});

// ../../node_modules/.pnpm/@smithy+types@3.1.0/node_modules/@smithy/types/dist-es/stream.js
var init_stream = __esm({
  "../../node_modules/.pnpm/@smithy+types@3.1.0/node_modules/@smithy/types/dist-es/stream.js"() {
    "use strict";
  }
});

// ../../node_modules/.pnpm/@smithy+types@3.1.0/node_modules/@smithy/types/dist-es/streaming-payload/streaming-blob-common-types.js
var init_streaming_blob_common_types = __esm({
  "../../node_modules/.pnpm/@smithy+types@3.1.0/node_modules/@smithy/types/dist-es/streaming-payload/streaming-blob-common-types.js"() {
    "use strict";
  }
});

// ../../node_modules/.pnpm/@smithy+types@3.1.0/node_modules/@smithy/types/dist-es/streaming-payload/streaming-blob-payload-input-types.js
var init_streaming_blob_payload_input_types = __esm({
  "../../node_modules/.pnpm/@smithy+types@3.1.0/node_modules/@smithy/types/dist-es/streaming-payload/streaming-blob-payload-input-types.js"() {
    "use strict";
  }
});

// ../../node_modules/.pnpm/@smithy+types@3.1.0/node_modules/@smithy/types/dist-es/streaming-payload/streaming-blob-payload-output-types.js
var init_streaming_blob_payload_output_types = __esm({
  "../../node_modules/.pnpm/@smithy+types@3.1.0/node_modules/@smithy/types/dist-es/streaming-payload/streaming-blob-payload-output-types.js"() {
    "use strict";
  }
});

// ../../node_modules/.pnpm/@smithy+types@3.1.0/node_modules/@smithy/types/dist-es/transfer.js
var RequestHandlerProtocol;
var init_transfer = __esm({
  "../../node_modules/.pnpm/@smithy+types@3.1.0/node_modules/@smithy/types/dist-es/transfer.js"() {
    "use strict";
    (function(RequestHandlerProtocol2) {
      RequestHandlerProtocol2["HTTP_0_9"] = "http/0.9";
      RequestHandlerProtocol2["HTTP_1_0"] = "http/1.0";
      RequestHandlerProtocol2["TDS_8_0"] = "tds/8.0";
    })(RequestHandlerProtocol || (RequestHandlerProtocol = {}));
  }
});

// ../../node_modules/.pnpm/@smithy+types@3.1.0/node_modules/@smithy/types/dist-es/transform/client-payload-blob-type-narrow.js
var init_client_payload_blob_type_narrow = __esm({
  "../../node_modules/.pnpm/@smithy+types@3.1.0/node_modules/@smithy/types/dist-es/transform/client-payload-blob-type-narrow.js"() {
    "use strict";
  }
});

// ../../node_modules/.pnpm/@smithy+types@3.1.0/node_modules/@smithy/types/dist-es/transform/no-undefined.js
var init_no_undefined = __esm({
  "../../node_modules/.pnpm/@smithy+types@3.1.0/node_modules/@smithy/types/dist-es/transform/no-undefined.js"() {
    "use strict";
  }
});

// ../../node_modules/.pnpm/@smithy+types@3.1.0/node_modules/@smithy/types/dist-es/transform/type-transform.js
var init_type_transform = __esm({
  "../../node_modules/.pnpm/@smithy+types@3.1.0/node_modules/@smithy/types/dist-es/transform/type-transform.js"() {
    "use strict";
  }
});

// ../../node_modules/.pnpm/@smithy+types@3.1.0/node_modules/@smithy/types/dist-es/uri.js
var init_uri = __esm({
  "../../node_modules/.pnpm/@smithy+types@3.1.0/node_modules/@smithy/types/dist-es/uri.js"() {
    "use strict";
  }
});

// ../../node_modules/.pnpm/@smithy+types@3.1.0/node_modules/@smithy/types/dist-es/util.js
var init_util = __esm({
  "../../node_modules/.pnpm/@smithy+types@3.1.0/node_modules/@smithy/types/dist-es/util.js"() {
    "use strict";
  }
});

// ../../node_modules/.pnpm/@smithy+types@3.1.0/node_modules/@smithy/types/dist-es/waiter.js
var init_waiter = __esm({
  "../../node_modules/.pnpm/@smithy+types@3.1.0/node_modules/@smithy/types/dist-es/waiter.js"() {
    "use strict";
  }
});

// ../../node_modules/.pnpm/@smithy+types@3.1.0/node_modules/@smithy/types/dist-es/index.js
var init_dist_es = __esm({
  "../../node_modules/.pnpm/@smithy+types@3.1.0/node_modules/@smithy/types/dist-es/index.js"() {
    "use strict";
    init_abort();
    init_auth2();
    init_blob_payload_input_types();
    init_checksum();
    init_client();
    init_command();
    init_connection();
    init_crypto();
    init_encode();
    init_endpoint();
    init_endpoints();
    init_eventStream();
    init_extensions();
    init_http();
    init_httpHandlerInitialization();
    init_identity2();
    init_logger();
    init_middleware();
    init_pagination();
    init_profile();
    init_response();
    init_retry();
    init_serde();
    init_shapes();
    init_signature();
    init_stream();
    init_streaming_blob_common_types();
    init_streaming_blob_payload_input_types();
    init_streaming_blob_payload_output_types();
    init_transfer();
    init_client_payload_blob_type_narrow();
    init_no_undefined();
    init_type_transform();
    init_uri();
    init_util();
    init_waiter();
  }
});

export {
  EndpointURLScheme,
  AlgorithmId,
  SMITHY_CONTEXT_KEY,
  IniSectionType,
  init_dist_es
};
